plinth/modules/matrixsynapse/views.py

```diff
@@ -28,13 +28,19 @@
 from plinth.modules import matrixsynapse
 from plinth.forms import DomainSelectionForm
 from plinth.utils import get_domain_names
+from django.contrib import messages
+from django.utils.translation import ugettext as _
+from plinth.utils import YAMLFile
+
+from .forms import MatrixSynapseServiceForm
 
 
 class SetupView(FormView):
     """Show matrix-synapse setup page."""
     template_name = 'matrix-synapse-pre-setup.html'
     form_class = DomainSelectionForm
     success_url = reverse_lazy('matrixsynapse:index')
+    form_class = MatrixSynapseServiceForm
```

Comment:
You're only changing the SetupView, but not the ServiceView, which I think is more important, so that users can change their mind and enable/disable even after it's installed. See the current code, there the option is only visible in the ServiceView (arguably, would be nice to decide before install, too).